src/storage/heap_file.c

```diff
@@ -20091,7 +20201,8 @@ heap_update_adjust_recdes_header (THREAD_ENTRY * thread_p, HEAP_OPERATION_CONTEX
 	}
     }
 
-  if (is_mvcc_op)
+#if defined (SERVER_MODE)
+  if (is_mvcc_class && !is_update_in_place)
```

Comment:
I hope to keep `is_mvcc_op` parameter and have
```C
  if (is_mvcc_op)
    {
      assert (is_mvcc_class && !is_update_in_place);
      ...
```

The current patch is correct but I'd like to avoid a mistake when someone modifies the condition to decide `is_mvcc_op` from `heap_update_logical`. 

Alternative is to add comments to warn the potential risk.